ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -22,6 +23,11 @@ S3Connection::S3Connection(const Aws::Auth::AWSCredentials &credentials,
   const std::string &objPrefix)
     : _s3CrtClient(credentials, config), _bucketName(bucketName), _objectPrefix(objPrefix)
 {
+    /* Confirm that we can access the bucket, else fail. */
+    bool exists;
+    int ret = BucketExists(exists);
+    if (ret != 0 || !exists)
+        throw std::invalid_argument(_bucketName + " : No such bucket.");
```

Comment:
We can print `ret` just in case and the `bucketName` to give all the relevant info back to the caller.

Reply:
I have thought about it, we will need to do a cleanup/hardening of the code, WT-8855. We will work out in that ticket a general paradigm on how we return the error code and error messages for the connection class methods (including the constructor). Let's address this issue at that moment. We will have more time to have a conversation around it.